wizeline/WLEmptyState
# Setup snapshot tests

#### What does this PR do? Any background context you want to provide?

It adds the setup of [FBSnapshotTestCase](https://github.com/uber/ios-snapshot-test-case) to test the SampleViewController view within the example project.

When the record mode is turned on in the snapshot tests, It captures a reference image inside the folder specified by the environment variable `FB_REFERENCE_IMAGE_DIR`. 
Once it is saved and the record mode is turned off, it is ready to execute UI test by comparing the saved image to the view generated by the controller. If there is a difference between these two, it will fail the test and a difference image will be stored in the folder specified by the environment variable `IMAGE_DIFF_DIR`.

For this project, the setup is done to be stored like the following:

| Variable  | Folder |
| ------------- | ------------- |
| FB_REFERENCE_IMAGE_DIR  | Example/Tests/ReferenceImages  |
| IMAGE_DIFF_DIR  | Example/Tests/FailureDiffs  |

#### Where should the reviewer start?

 - SnapshotTests.swift

#### How should this be manually tested?

Just run the test included in the class `SnapshotTests`.

Right now reference images for iPhone XR and iPhone X executions are being included. Make sure to select any of these 2 simulators to run the tests and check. Otherwise, you can turn on the record mode and generate a new reference for any other device. 

Although, please be aware that FBSnapshotTestCase generates **only one** reference image per scale. Therefore, if you choose some other simulator with the same scales as the previous (i.e. 2x and 3x respectively), it will not only fail the test and save the diff, but also it will ask to record again and overwrite the previous reference due to the image size differences.

#### What are the relevant tickets?

[#8](https://github.com/wizeline/WLEmptyState/issues/8)

#### Screenshots (if appropriate)

Testing the view: 
 
![screen shot 2019-02-04 at 9 43 32 pm](https://user-images.githubusercontent.com/40211551/52253675-0fcdd600-28cf-11e9-8eb4-8919509fb450.png)
